### PR TITLE
feat: Sync ray 2.4 parquet datasource

### PR DESCRIPTION
**Pull request by kukushking**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 7c3d8aeefbcf538e5cc2f84bcc14cadfdffa6dcb
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a536d014-ac20-4d18-a0d7-94783e145e72%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 7c3d8aeefbcf538e5cc2f84bcc14cadfdffa6dcb
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d47b6f3e-16a4-4ae8-bdb1-5b2cefa98a86%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: bc48b12b1e601990f217f7639d3b831a579ea39e
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=87652652-6dc3-4f53-b988-424e106a4185%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: bc48b12b1e601990f217f7639d3b831a579ea39e
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ff050add-2d8e-4f34-a8d1-c0b1bc81b9f1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: pyproject.toml [hunk 1]
```diff
@@ -59,6 +59,7 @@ deltalake = { version = ">=0.6.4,<0.10.0", optional = true }
 
 # Distributed
 modin = { version = "^0.20.1", optional = true }
+s3fs = { version = "0.4.2", optional = true }
```
Comment (jaidisido) [inline pyproject.toml:62]: Why would we need s3fs as a required dependency when installing Modin? Not even Modin itself requires it

Comment (kukushking) [reply pyproject.toml:62]: It's a fair default for Modin to keep it optional because you might not be using s3 at all. I added this here because modin load tests are failing with missing s3fs

Comment (jaidisido) [reply pyproject.toml:62]: But we only need it for our load tests, none of our actual awswrangler methods actually use s3fs. In the load test, it's only modin specific tests (i.e. no awswrangler involved that use it). So I don't think we want to add a dependency on users if we don't need it

Comment (kukushking) [reply pyproject.toml:62]: Ok, agreed. I will put it into the infra instead

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: f4cded3d13f3f3263770104ac5ee2c22ef6a0553
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5451f21d-e310-4888-8a28-2d3ba482ba49%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by jaidisido — COMMENTED** [review]
Is the change to the poetry.lock still required?

**Comment by kukushking** [comment]
@jaidisido yes, because of the typing stubs.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: be4e71ced7deee2856741d7d2b993de180bbc7e7
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8dff1ea0-142e-4ec0-96e8-cd90f5e50d2c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by jaidisido — APPROVED** [review]
